### PR TITLE
Add pull to refresh

### LINKED ISSUE #19: Add pull to refresh

### PR DESCRIPTION
**Pull request by deweytyl**
Double commit for "Fix formatting and refactoring" and merge because I screwed up and did a `git commit --amend` after I had pushed to this repo.

Closes #19

### CODE REVIEW

File: SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m [hunk 1]
```diff
@@ -56,35 +43,34 @@ - (void)viewDidLoad
         //NSLog(@"Received Category: %@", self.recievedCategory);
```
Comment (alexthemitchell) [inline SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m]: extraneous comment

File: SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m [hunk 2]
```diff
@@ -56,35 +43,34 @@ - (void)viewDidLoad
...
+    } else if ([self.newsCategory isEqualToString:@"Favorites"]) {
         // Do nothing on favorites.
         self.allArticlesArray =  [[DataModel sharedModel] savedArticles];
         [self.tableView reloadData];
-    }
-    else {
         
+    } else {
         //Check cache content of category.
         NSMutableArray *categoryCache = [[Cache sharedCacheModel] loadArchivedObjectWithFileName:self.recievedCategory];
```
Comment (alexthemitchell) [inline SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m]: How does this compare to the steps in lines 51-57?

File: SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m [hunk 3]
```diff
@@ -92,152 +78,165 @@ - (void)viewWillAppear:(BOOL)animated {
...
+        } else {
             NSLog(@"I am sad!: %@", [error description]);
         }
+        
         self.isFetchingArticles = NO;
+        [self.refreshControl endRefreshing];
     }];
 }
 
 - (void)fetchCategoryArticles {
```
Comment (alexthemitchell) [inline SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m]: What's the difference between this and `-(void)fetchArticles`?

File: SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m [hunk 4]
```diff
@@ -92,152 +78,165 @@ - (void)viewWillAppear:(BOOL)animated {
...
+        cell.categoryIdentifier.backgroundColor = category.readBarColor;
+        
+    } else {
         [cell.articleTitle setFont:[UIFont fontWithName:@"ProximaNova-Regular" size:18]];
+        
+        cell.articleTitle.textColor = [UIColor blackColor];
         cell.articleDetails.textColor = [UIColor blackColor];
-        cell.categoryIdentifier.backgroundColor = [[[[NewsCategories sharedCategories] categoriesByName] objectForKey:[[self.allArticlesArray objectAtIndex:indexPath.row] category]] color];
+        
+        cell.backgroundColor = [UIColor whiteColor];
```
Comment (alexthemitchell) [inline SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m]: redundant change of `cell.backgroundColor`? See line 178.

File: SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m [hunk 5]
```diff
@@ -92,152 +78,165 @@ - (void)viewWillAppear:(BOOL)animated {
...
         return [self articleCellForIndexPath:indexPath];
     } else {
         return [self loadingCell];
     }
 }
 
-- (UITableViewCell *)loadingCell
-{
-    UITableViewCell *cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:nil];
+- (UITableViewCell *)loadingCell {
```
Comment (alexthemitchell) [inline SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m]: Is this necessary? I don't think I've ever actually made a separate cell for my `UIActivtyIndicatorView`. See https://github.com/alexthemitchell/Foundit/blob/master/Found%20It!/Found%20It!/FoundTableViewController.m#L52.

File: SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m [hunk 6]
```diff
@@ -92,152 +78,165 @@ - (void)viewWillAppear:(BOOL)animated {
...
     ArticleCell *cell = (ArticleCell *) [tableView cellForRowAtIndexPath:indexPath];
+    Article *article = [self.allArticlesArray objectAtIndex:indexPath.row];
+    NewsCategory *category = [[[NewsCategories sharedCategories] categoriesByName]
+                                 objectForKey:article.category];
     
-    cell.backgroundColor = [[[[NewsCategories sharedCategories] categoriesByName] objectForKey:[[self.allArticlesArray objectAtIndex:indexPath.row] category]] highlightedColor];
+    cell.backgroundColor = category.highlightedColor;
     
-    [[DataModel sharedModel] markArticleAsRead:[self.allArticlesArray objectAtIndex:indexPath.row]];
+    [[DataModel sharedModel] markArticleAsRead:article];
```
Comment (alexthemitchell) [inline SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m:122]: should `read` be a property on `Article`?

File: SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m (hunk 5 above)
Comment (deweytyl) [inline SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m]: This cell is inserted at the bottom of the table view. It's for loading the next "page" of articles, not at the top.

Comment (alexthemitchell) [inline SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m]: Cool. Never mind, then.

File: SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m (hunk 2 above)
Comment (deweytyl) [inline SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m]: Same thing but this block uses fetchCategoryArticles and the first uses fetchArticles. Same thing happens all over this file. This all goes back to two different methods in DataModel. Going forward I would want to move that distinction behind a fetchArticlesForCategory method in the data model that does something special (`get_recent_posts`) for the "News" category, and this VC should always call the same method. But for now I'll refactor this a little to make it less repetitive.

Comment (deweytyl) [inline SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m]: Same thing but this block uses fetchCategoryArticles and the first uses fetchArticles. Same thing happens all over this file. This all goes back to two different methods in DataModel. Going forward I would want to move that distinction behind a fetchArticlesForCategory method in the data model that does something special (`get_recent_posts`) for the "News" category, and this VC should always call the same method. But for now I'll refactor this a little to make it less repetitive.

Comment (deweytyl) [inline SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m]: Same thing but this block uses fetchCategoryArticles and the first uses fetchArticles. Same thing happens all over this file. This all goes back to two different methods in DataModel. Going forward I would want to move that distinction behind a fetchArticlesForCategory method in the data model that does something special (`get_recent_posts`) for the "News" category, and this VC should always call the same method. But for now I'll refactor this a little to make it less repetitive.

File: SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m (hunk 3 above)
Comment (deweytyl) [inline SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m]: See earlier note. Just the method called on DataModel.

File: SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m (hunk 2 above)
Comment (alexthemitchell) [inline SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m]: Perfect. Update that and then this PR LGTM.

File: SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m (hunk 6 above)
Comment (deweytyl) [inline SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m:122]: Probably, but that refactoring will come later

**Pull request merged**